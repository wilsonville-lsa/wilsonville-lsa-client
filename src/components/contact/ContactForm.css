.Contact {
  width: 70vw;
  margin: auto;
  padding: 10vh;

  .red {
    color: red;
  }

  .green {
    color: green;
  }

  aside {
    float: left;
    text-align: left;

    p {
      padding: 0;
      margin: 1vh;
    }
  }

  section {
    float: right;
    width: 40vw;
    clear: both;

    p {
      text-align: center;
    }

    form {
      text-align: left;
      
      label {
        display: block;
        margin-top: 2vh;
        color: green;
      }

      input {
        margin-left: 5rem;
      }

      textarea {
        vertical-align: text-top;
        margin-left: 3.5rem;
      }

      button {
        margin-top: 5vh;
        padding: 0.5rem;
        color: green;
      }

      button:hover {
        color: rgb(9, 177, 9);
      }
    }
  }

  h3 {
    margin: 0;
    margin-bottom: 5vh;
    text-align: center;
  }
}