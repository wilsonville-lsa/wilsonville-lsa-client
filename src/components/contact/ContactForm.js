/* eslint-disable react/no-unescaped-entities */
import React from 'react';
import PropTypes from 'prop-types';
import styles from './ContactForm.css';

function ContactForm({ message, handleChange, handleSubmit, error, confirmation }) {
  return (
    <main className={styles.Contact}>
      <aside>
        <h3>Mail</h3>
        <p>Wilsonville Baha'i Community</p>
        <p>PO Box 2641</p>
        <p>Wilsonville, OR 97070</p>
      </aside>

      <section>

        {error && <p className={styles.red}>{error}</p>}

        {confirmation && <p className={styles.green}>Your message has been sent! We will respond shortly. A confirmation email has been sent to the provided email. If you have not received a confirmation, please re-send your message. Thank you!</p>}
        
        <form onSubmit={handleSubmit.bind(null, message)}>
          <p>If you wish to connect with your local Baha'i community, please use the form below to send us a message.</p>
          <label>
            Name:<span className={styles.red}>*</span>
            <input name="name" type="text" onChange={handleChange} required/>
          </label>
          <label>
            Email:<span className={styles.red}>*</span>
            <input name="email" type="text" onChange={handleChange} required/>
          </label>
          <label>
            Message:<span className={styles.red}>*</span>
            <textarea name="text" onChange={handleChange} required/>
          </label>
          <label>
            <button type="submit">Submit</button>
          </label>
        </form>
      </section>
    </main>
  );
}

ContactForm.propTypes = {
  message: PropTypes.object.isRequired,
  handleChange: PropTypes.func.isRequired,
  handleSubmit: PropTypes.func.isRequired,
  error: PropTypes.string,
  confirmation: PropTypes.bool.isRequired
};

export default ContactForm;
