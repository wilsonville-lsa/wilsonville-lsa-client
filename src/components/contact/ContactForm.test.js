import React from 'react';
import renderer from 'react-test-renderer';
import ContactForm from './ContactForm';

describe('Contact Form', () => {
  it('matches snapshot for ContactForm', () => {
    const props = {
      message: {},
      handleChange: jest.fn(),
      handleSubmit: jest.fn(),
      error: '',
      confirmation: false
    };

    const tree = renderer.create(
      <ContactForm {...props} />
    );

    expect(tree).toMatchSnapshot();
  });

  it('matches snapshot for ContactForm with confirmation', () => {
    const props = {
      message: {
        name: 'Inquirer',
        email: 'inquirer@email.com',
        text: 'Inquiry inquiring about an inquiry'
      },
      handleChange: jest.fn(),
      handleSubmit: jest.fn(),
      error: '',
      confirmation: true
    };

    const tree = renderer.create(
      <ContactForm {...props} />
    );

    expect(tree).toMatchSnapshot();
  });

  it('matches snapshot for ContactForm with error', () => {
    const props = {
      message: {
        name: 'Inquirer',
        email: 'inquirer@email.com',
        text: 'Inquiry inquiring about an inquiry'
      },
      handleChange: jest.fn(),
      handleSubmit: jest.fn(),
      error: 'Invalid email address',
      confirmation: false
    };

    const tree = renderer.create(
      <ContactForm {...props} />
    );

    expect(tree).toMatchSnapshot();
  });
});
