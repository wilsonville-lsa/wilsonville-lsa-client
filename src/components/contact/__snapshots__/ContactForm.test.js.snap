// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact Form matches snapshot for ContactForm 1`] = `
<main
  className="Contact"
>
  <aside>
    <h3>
      Mail
    </h3>
    <p>
      Wilsonville Baha'i Community
    </p>
    <p>
      PO Box 2641
    </p>
    <p>
      Wilsonville, OR 97070
    </p>
  </aside>
  <section>
    
    <form
      onSubmit={[Function]}
    >
      <p>
        If you wish to connect with your local Baha'i community, please use the form below to send us a message.
      </p>
      <label>
        Name:
        <span
          className="red"
        >
          *
        </span>
        <input
          name="name"
          onChange={[MockFunction]}
          required={true}
          type="text"
        />
      </label>
      <label>
        Email:
        <span
          className="red"
        >
          *
        </span>
        <input
          name="email"
          onChange={[MockFunction]}
          required={true}
          type="text"
        />
      </label>
      <label>
        Message:
        <span
          className="red"
        >
          *
        </span>
        <textarea
          name="text"
          onChange={[MockFunction]}
          required={true}
        />
      </label>
      <label>
        <button
          type="submit"
        >
          Submit
        </button>
      </label>
    </form>
  </section>
</main>
`;

exports[`Contact Form matches snapshot for ContactForm with confirmation 1`] = `
<main
  className="Contact"
>
  <aside>
    <h3>
      Mail
    </h3>
    <p>
      Wilsonville Baha'i Community
    </p>
    <p>
      PO Box 2641
    </p>
    <p>
      Wilsonville, OR 97070
    </p>
  </aside>
  <section>
    
    <p
      className="green"
    >
      Your message has been sent! We will respond shortly. A confirmation email has been sent to the provided email. If you have not received a confirmation, please re-send your message. Thank you!
    </p>
    <form
      onSubmit={[Function]}
    >
      <p>
        If you wish to connect with your local Baha'i community, please use the form below to send us a message.
      </p>
      <label>
        Name:
        <span
          className="red"
        >
          *
        </span>
        <input
          name="name"
          onChange={[MockFunction]}
          required={true}
          type="text"
        />
      </label>
      <label>
        Email:
        <span
          className="red"
        >
          *
        </span>
        <input
          name="email"
          onChange={[MockFunction]}
          required={true}
          type="text"
        />
      </label>
      <label>
        Message:
        <span
          className="red"
        >
          *
        </span>
        <textarea
          name="text"
          onChange={[MockFunction]}
          required={true}
        />
      </label>
      <label>
        <button
          type="submit"
        >
          Submit
        </button>
      </label>
    </form>
  </section>
</main>
`;

exports[`Contact Form matches snapshot for ContactForm with error 1`] = `
<main
  className="Contact"
>
  <aside>
    <h3>
      Mail
    </h3>
    <p>
      Wilsonville Baha'i Community
    </p>
    <p>
      PO Box 2641
    </p>
    <p>
      Wilsonville, OR 97070
    </p>
  </aside>
  <section>
    <p
      className="red"
    >
      Invalid email address
    </p>
    <form
      onSubmit={[Function]}
    >
      <p>
        If you wish to connect with your local Baha'i community, please use the form below to send us a message.
      </p>
      <label>
        Name:
        <span
          className="red"
        >
          *
        </span>
        <input
          name="name"
          onChange={[MockFunction]}
          required={true}
          type="text"
        />
      </label>
      <label>
        Email:
        <span
          className="red"
        >
          *
        </span>
        <input
          name="email"
          onChange={[MockFunction]}
          required={true}
          type="text"
        />
      </label>
      <label>
        Message:
        <span
          className="red"
        >
          *
        </span>
        <textarea
          name="text"
          onChange={[MockFunction]}
          required={true}
        />
      </label>
      <label>
        <button
          type="submit"
        >
          Submit
        </button>
      </label>
    </form>
  </section>
</main>
`;
